zulipterminal/model.py

```diff
@@ -885,6 +885,17 @@ def is_muted_topic(self, stream_id: int, topic: str) -> bool:
         topic_to_search = (stream_name, topic)
         return topic_to_search in self._muted_topics
 
+    def get_current_topic_from_message(
+        self, message_id: int
+    ) -> Union[Tuple[int, str], None]:
+        topic_msg_ids = self.index["topic_msg_ids"]
```

Comment:
Generally the index is not guaranteed to be complete (fully fetched) or in sync with all other data we get from the server (possibly including unreads), and that includes this element of it.

In particular, here we only update this data when we fetch messages for a topic narrow, typically when switching to it.

I think this should be OK here, but it may be worth adding a docstring and test to make this explicit?

In terms of implementation, using the main `index["messages"]` may be simpler?